atc/db/team.go

```diff
@@ -1286,6 +1296,47 @@ func saveResource(tx Tx, resource atc.ResourceConfig, pipelineID int) (int, erro
 	return resourceID, nil
 }
 
+func existingResourceConfig(tx Tx, name string, pipelineID int) (atc.ResourceConfig, string, error) {
+	var (
+		configBlob, nonce sql.NullString
+		oldType           string
+	)
+	row := psql.Select("config", "type", "nonce").
+		From("resources").
+		Where(sq.Eq{
+			"name":        name,
+			"pipeline_id": pipelineID,
+		}).
+		RunWith(tx).
+		QueryRow()
+	err := row.Scan(&configBlob, &oldType, &nonce)
+	if err != nil && err != sql.ErrNoRows {
+		if err == sql.ErrNoRows {
```

Comment:
We will never reach here because there is the `&& err != sql.ErrNoRows` check right above it